Added content to "Setting up a new project" page (#180)
pharo-open-documentation/pharo-wiki: General/SettingUpANewProject.md

## Step 2. Create a package and a class

Let us make our first package. In your newly made image in Pharo Launcher, look at the menu on the top. Choose a tools card and from there open System Browser.

![](SettingUp_SystemBrowser.png)

System Browser is split into five parts. The most left one, shows all the packages that already exist. To create a new one, press the right click of a mouse on this part, and choose the option New Package. 

![](SettingUp_NewPackage.png)

> Choosing it, a new small window will open, where you give a name to the package you are creating.

Reviewer: ```suggestion
Choosing it, a new small window will open, where you give a name to the package you are creating. Type the name *Counter* and press ok.
```

> We will show this by following given example. Type the name *Counter* in Window for creating a new package and press ok.

Reviewer: ```suggestion

```

To create a class, first you need to choose the package in which it will be located. Then look at the bottom part of the System Browser in it there is a template for creating a class. It gives us all the information we need: superclass, variables, package. Creating a new class is as easy as filling out this template. Continuing with our example, we will create a new class *MyCounter*. The superclass is *Object*, so we will not change that. But we will change the name of the class. You will notice that saving our class, the second left part of SystemBrowser is no longer empty.

![](SettingUp_AddingClass.png)

## Step 3. Write some failing tests

> If we want to follow the TDD (test driven development), we need to create tests first. All tests are stored in a test class, which is a subclass of *TestCase*. By useing the previous step, make Counter-Tests package and CounterTest class.

Reviewer: ```suggestion
If we want to follow [TDD (test driven development)](https://fr.wikipedia.org/wiki/Test_driven_development), we first need to create tests. All tests are stored in a test class, which is a subclass of *TestCase*. This class can be generated by doing a right-click on the class name and selecting `Jump to test class`. This option will generate a test class in a new package `Counter-Tests`.
```

In case there are no protocols to choose from (this happens in cases when we are making our first test), then we need to add a new one. This is done by pressing a right click on *instance side*, going to New protocol option and picking a protocol you need (in our case tests).

![](SettingUp_NewProtocol.png)

After we have choosen a protocol, our bottom part of System Browser will change and show a template for making methods (each test is a separate method). When giving a name it is important to put the word *test* first, so that Pharo knows that what we are making is in fact a test.

In our example, a counter will be incrementing and decrementing some value. You will notice that we are using methods which have not been created yet and that each test you write will fail (be red) when you run it. 

We can make these tests: 

**Creating MyCounter object:** 
We want to check if the object we created is not equal to nil, that is why we are using *self deny:* message. This will to a check if the statement we send is true and if it is return false. 

> ![](SettingUp_FirstTestObject.png)

Reviewer: For methods to implement I would use code snippet instead of screenshots for two reasons:
- Code snippet can be copy/paste
- Screenshots can become obsolete with different Pharo versions, so I tend to limit their number
